### ISSUE TITLE
Enhance interface by making search box always visible

### ISSUE THREAD

**Issue by t4777sd** [keepassxreboot/keepassx#15]
Search box is NOT visible by default. Searching passwords is probably the most common task in keepassx. Why does it require an extra click to toggle the search? That is likely one of the most used interface features and it is hidden by default which creates unnecessary friction to use one of the most used aspects of the software.

**Comment by droidmonkey** [comment]
I have completed my first revision of this feature request. I added the ability to hit ESCAPE to cancel the search. When the search box is cleared that also cancels the search. Searches now transcend between databases.

Things still left to sort out:
- Desired behavior when moving between databases
- Search icon produces context menu to select search options
- Implement a global search (will take a lot of work)
